src/base/bittorrent/torrentimpl.cpp

```diff
@@ -1516,6 +1516,16 @@ qlonglong TorrentImpl::nextAnnounce() const
     return lt::total_seconds(m_nativeStatus.next_announce);
 }
 
+// in order to produce floating-point numbers using `std::chrono::duration_cast`,
+// we should use `qreal` as `Rep` to define the `months` duration
+using months = std::chrono::duration<qreal, std::chrono::months::period>;
+
+qreal TorrentImpl::popularity() const
+{
+    const auto activeMonths = std::chrono::duration_cast<months>(m_nativeStatus.active_duration).count();
```

Comment:
I think it can be simplified to this:
```suggestion
    const qreal activeMonths = lt::duration_cast<std::chrono::months>(m_nativeStatus.active_duration).count();
```
Note that
* `lt::duration_cast` is preferred since you are dealing with `active_duration` which is `lt::seconds` type. ps. don't need to remind me that libtorrent redefined it from std::chrono.
* libtorrent hasn't define `month` type so `std::chrono::months` is preserved.
* I would prefer conversion to `qreal` is done at the last step of calculation as to avoid (potential) loss of precision issues.

Reply:
Well ignore my above comment. Seems you really need to preform the conversion in `qreal` type.